test/remote/gateways/remote_shift4_test.rb

```diff
@@ -0,0 +1,247 @@
+require 'test_helper'
+
+class RemoteShift4Test < Test::Unit::TestCase
+  def setup
+    @gateway = Shift4Gateway.new(fixtures(:shift4))
+
+    @amount = 5
+    @invoice = '3333333309'
+    @credit_card = credit_card('4000100011112224')
+    @declined_card = credit_card('4000300011112220')
+    @declined_card.number = '400030001111220'
+    @options = {
+      order_id: '1',
+      company_name: 'Spreedly',
+      billing_address: address,
+      description: 'Store Purchase',
+      "tax": 1,
+      "tip": '',
+      "total": 5.00,
+      numeric_id: 24,
+      invoice: rand(4).to_s,
+      security_code: {
+        indicator: '1',
+        value: '4444'
+      }
+    }
+    @refund_options = @options.except(:invoice).merge({
+      invoice: @invoice,
+      notes: 'Transaction notes are added here',
+      billing_address: address,
+      stored_credential: {
+        indicator: '01',
+        usage_indicator: '01',
+        scheduled_indicator: '01',
+        transaction_id: 'yxx'
```

Comment:
Take a look at [remote_cyber_source_test.rb](https://github.com/activemerchant/active_merchant/blob/023f3e8685e763c8758292790e2962dbaffc0c1b/test/remote/gateways/remote_cyber_source_test.rb#L1009-L1052) for examples as to what `stored_credential` values we pass in the `options` hash.